tracker/tracker-vuex/src/index.ts

```diff
@@ -22,25 +22,52 @@ export default function(opts: Partial<Options> = {}) {
     }
     const encoder = new Encoder(sha1, 50);
     return store => {
-      store.subscribe((mutation, state) => {
-        if (options.filter(mutation, state)) {
-          try {
-            const { type } = mutation;
-            if (typeof type === 'string' && type) {
-              app.send(Messages.StateAction(type));
+      if (store.subscribe) {
+        // vuex        
+        store.subscribe((mutation, state) => {
+          if (options.filter(mutation, state)) {
+            try {
+              const { type } = mutation;
+              if (typeof type === 'string' && type) {
+                app.send(Messages.StateAction(type));
+              }
+              const _mutation = encoder.encode(
+                options.mutationTransformer(mutation)
+              );
+              const _state = encoder.encode(options.transformer(state));
+              const _table = encoder.commit();
+              for (let key in _table) app.send(Messages.OTable(key, _table[key]));
+              app.send(Messages.Vuex(_mutation, _state));
+            } catch {
+              encoder.clear();
             }
-            const _mutation = encoder.encode(
-              options.mutationTransformer(mutation)
-            );
-            const _state = encoder.encode(options.transformer(state));
-            const _table = encoder.commit();
-            for (let key in _table) app.send(Messages.OTable(key, _table[key]));
-            app.send(Messages.Vuex(_mutation, _state));
-          } catch {
-            encoder.clear();
           }
+        });
+        // pinia
+        if (store.$onAction) {
+          store.$onAction(({ name, store, args, after, onError }) => {
+            try {
+              let state = {}
+              state[store.$id] = store.$state
+              const mutation = { type: name, payload: JSON.parse(JSON.stringify(args)) }
+              if (options.filter(mutation, state)) {
+                const { type } = mutation;
+                if (typeof type === 'string' && type) {
+                  app.send(Messages.StateAction(type));
+                }
+                const _mutation = encoder.encode(options.mutationTransformer(mutation));
+                const _state = encoder.encode(options.transformer(state));
+                const _table = encoder.commit();
```

Comment:
Could you move the copy paste code to separate method/function to reuse it?
lines 30-42 and 54-65

Reply:
done